python/lsst/dax/ppdb/sql/_base.py

```diff
@@ -0,0 +1,475 @@
+# This file is part of dax_ppdb
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ["SqlBase"]
+
+import logging
+import os
+import sqlite3
+from collections.abc import MutableMapping
+from contextlib import closing
+from typing import Any
+
+import astropy
+import felis.datamodel
+import sqlalchemy
+import yaml
+from felis.datamodel import Schema as FelisSchema
+from lsst.dax.apdb import (
+    IncompatibleVersionError,
+    VersionTuple,
+    schema_model,
+)
+from lsst.dax.apdb.sql import ApdbMetadataSql, ModelToSql
+from lsst.dax.ppdb.ppdb import PpdbConfig
+from lsst.resources import ResourcePath
+from sqlalchemy.pool import NullPool
+
+from .config import PpdbSqlConfig
+
+_LOG = logging.getLogger(__name__)
+
+VERSION = VersionTuple(0, 1, 1)
+"""Version for the code defined in this module. This needs to be updated
+(following compatibility rules) when schema produced by this code changes.
+"""
+
+
+def _onSqlite3Connect(
+    dbapiConnection: sqlite3.Connection, connectionRecord: sqlalchemy.pool._ConnectionRecord
+) -> None:
+    # Enable foreign keys
+    with closing(dbapiConnection.cursor()) as cursor:
+        cursor.execute("PRAGMA foreign_keys=ON;")
+
+
+class SqlBase:
```

Comment:
`SqlBase` name is not super descriptive, it should at least mention PPDB, maybe `PpdbSqlBase`?